doc/dev_guide/deply_noobaa_on_minikube.md

```diff
@@ -0,0 +1,108 @@
+# Deploy Noobaa On Minikube Guide
+We will run the commands in the terminal, you may work with at least two tabs:
+1) For noobaa-core repository
+2) For noobaa-operator repository
+In each step, it is mentioned what tab you should use.
+
+## Configuration (First Time Only):
+
+### 1) Change Your Docker Resource Preferences:
+Click on docker logo -> preference… -> Resource (left choice in the menu: CPU: 6, Memory: 10 GB, SWAP: 1 GB, Disk: 80 GB
+
+### 2) Setting Your Minikube Configurations
+1) Install [minikube](https://minikube.sigs.k8s.io/docs/start/).
+2) Start your cluster
+```bash
+minikube start
+```
+3) Set configurations:
+```bash
+    minikube config set memory 8000
+    minikube config set cpus 5
+```
+
+## Instruction (Every time you open a tab):
+### 1) Check That Minikube Is Running
+```bash
+minikube status
+```
+
+### 2) Before Building The Images (Noobaa-Core Tab)
+1) Change the working directory to the local repository (in this example the local repository is inside `SourceCode` directory).
+```bash
+ cd ./SourceCode/noobaa-core
+```
+2) We will use minikube to run the tests. It is recommended to build all images on the minikube docker daemon. Configure your docker client to use minikube's docker run:
+```bash
+eval $(minikube docker-env)
+```
+### 3) Before Building The Images (Noobaa-Operator Tab)
+Change the working directory to the local repository (in this example the local repository is inside `SourceCode` directory).
+```bash
+ cd ./SourceCode/noobaa-operator
+```
+In order to build the CLI and the operator image run the following:
+```bash
+. ./devenv.sh
+```
+_Note: the file `devenv.sh` contains the command `eval $(minikube docker-env)`. We run the command `eval $(minikube docker-env)` prior to an image build (whether from noobaa core repository or noobaa operator repository)._
+### 4) Build Operator Images (Noobaa-Operator Tab)
+```bash
+make all
+```
+This will build the following:
+* noobaa-operator image with tag `noobaa/noobaa-operator:<major.minor.patch>` (for example: `noobaa/noobaa-operator:5.13.0`). this tag is used by default when installing with the CLI.
+* noobaa CLI. The `devenv.sh` script is setting an alias `nb` to run the local build of the CLI.
+
+### 5) Build Core Images (Noobaa-Core Tab)
+Run the following to build noobaa core image with the desired tag to build the tester image:
+```bash
+make noobaa
+```
+Change the tag name  `noobaa:latest noobaa-core:<tag-name>`, for example: 
+```bash
+docker tag noobaa:latest noobaa-core:my-deploy
+```
```

Comment:
a small suggestion, can we add a step to push the image to the docker repository ( quay.io or docker hub) and use that image for the build, so that we do not need to create an image for noobaa core and operator if there is no change in code?
`docker tag noobaa-core:<tag-name> quay.io/noobaa-test/noobaa-core:<tag-name>`
`docker push quay.io/noobaa-test/noobaa-core:<tag-name>`
nb install --mini --noobaa-image='quay.io/noobaa-test/noobaa-core:<tag-name>'
like core we can push operartor image and use that in noobaa install command 
`nb install --mini --noobaa-image='quay.io/noobaa-test/noobaa-core:<tag-name>' --operator-image='quay.io/noobaa-test/noobaa-operator:<tag-name> -n noobaa`